src/main/java/org/broadinstitute/hellbender/tools/spark/sv/evidence/XGBoostEvidenceFilter.java

```diff
@@ -0,0 +1,609 @@
+        }
+        return listItr.next();
+    }
+
+    private boolean isValidated( final List<BreakpointEvidence> evList ) {
+        for ( final BreakpointEvidence ev : evList ) {
+            if ( ev.isValidated() ) return true;
+        }
+        return false;
+    }
+
+    boolean anyPassesFilter(final List<BreakpointEvidence> evidenceList) {
+        for(final BreakpointEvidence evidence : evidenceList) {
+            final double evidenceGoodProbability = predictor.predictSingle(getFeatures(evidence));
+            if(evidenceGoodProbability > thresholdProbability) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Compute features vector for a piece of BreakpointEvidence
+     */
+    @VisibleForTesting
+    EvidenceFeatures getFeatures(final BreakpointEvidence evidence) {
+        // create new struct for these two, use CigarOperator to update if instanceof ReadEvidence
+        final CigarQualityInfo cigarQualityInfo = new CigarQualityInfo(evidence);
+        final double evidenceType = evidenceTypeMap.get(evidence.getClass().getSimpleName());
+        final double mappingQuality = (double)getMappingQuality(evidence);
+        final double templateSize = getTemplateSize(evidence);
+
+        // calculate these similar to BreakpointDensityFilter, but always calculate full totals, never end early.
+        final CoverageScaledOverlapInfo individualOverlapInfo = getIndividualOverlapInfo(evidence);
+        final CoverageScaledOverlapInfo clusterOverlapInfo = getClusterOverlapInfo(evidence);
+
+        // calculate properties related to overlap of intervals on the reference genome
+        final double referenceGapOverlap = getGenomeIntervalsOverlap(evidence, genomeGaps);
+        final double umapS100 = getGenomeIntervalsOverlap(evidence, umapS100Mappability);
+
+        return new EvidenceFeatures(
+            new double[]{
+                cigarQualityInfo.basesMatched, cigarQualityInfo.referenceLength, evidenceType, mappingQuality, templateSize,
+                individualOverlapInfo.numOverlap, individualOverlapInfo.overlapMappingQuality,
+                individualOverlapInfo.meanOverlapMappingQuality, individualOverlapInfo.numCoherent,
+                individualOverlapInfo.coherentMappingQuality,
+                clusterOverlapInfo.numOverlap, clusterOverlapInfo.overlapMappingQuality,
+                clusterOverlapInfo.meanOverlapMappingQuality, clusterOverlapInfo.numCoherent,
+                clusterOverlapInfo.coherentMappingQuality,
+                referenceGapOverlap, umapS100
+            }
+        );
+    }
+
+    private int getMappingQuality(final BreakpointEvidence evidence) {
+        // Note: return "max" mapping quality for non-ReadEvidence. Reasoning: some features depend on sum or average of
+        // read qualities. Non-ReadEvidence isn't *bad* per se, so give it a good score.
+        return evidence instanceof BreakpointEvidence.ReadEvidence ?
```

Comment:
`instanceof` is a potential code smell.  Why not just implement `getMappingQuality` for other types of evidence and set it to 60?

Reply:
Or set is to `Optional.empty` and let this method translate that to 60.

Reply:
I've removed "instanceof " in favor of polymorphism with base class returning null by default and classes where values are sensible returning good values.
I've used null instead of Optional.empty because that's the style throughout the project.